### PR TITLE
Remove Batch tests; moved to standalone Batch TCK: https://github.com…

### LINKED ISSUE #574: Consume Jakarta Batch tests from batch-tck rather than duplicating within this project.
## BASIC BACKGROUND

The Jakarta Batch tests in this repository are rooted off of path:  https://github.com/eclipse-ee4j/jakartaee-tck/tree/master/src/com/ibm/jbatch/tck.

These essentially duplicate the same exact tests in https://github.com/eclipse-ee4j/batch-tck.
This is a legacy of the way in which the tests were originally provided to CTS.   

There may be some differences (I haven't checked recently) related to logging and assertions, but the core batch semantics of the tests are (should be) exactly the same.

It seems like a step forward would be to update the build in this project and pull in the artifacts from the 'batch-tck' maven coordinates, but to otherwise execute the tests, within jakartee-tck automation, in the exact way they're executing now.

## ALTERNATIVE

Lurking in the background, there's the idea of making the batch-tck project more flexible and useful, to the point where there's no need to include the batch tests in this  jakartee-tck project.

E.g. we talked about some enhancements:  https://ondro.inginea.eu/index.php/possible-ways-to-use-arquillian-in-jakarta-ee-tcks/

If that seems like the preferred approach, to decouple from this repo, then well, possibly we should just focus on that. 
Although pulling in the tests from batch-tck seems easier if we're at the point where we want to add some new batch tests, and don't want to duplicate.

### SOME MORE DETAILS

If we divide up the artifacts into:
1. test classes (with test methods), test utilities
2. batch artifacts (things that directly implement the batch API) 

I'd expect 2. to be exactly the same with the differences in 1.   (I could be wrong though, if some of the logging diffs are present in 2.).

### PR DESCRIPTION
**Pull request by scottkurz**
…/eclipse-ee4j/batch-tck

Signed-off-by: Scott Kurz <skurz@us.ibm.com>

---
name: Pull Request
about: Create a pull request for a Platform TCK change
title: ''
labels: ''
assignees: ''

---

**Fixes Issue**
Fixes #574 

**Related Issue(s)**
Specify any related issue(s) links.

**Describe the change**
Remove the Batch tests now covered via standalone TCK: 
https://github.com/eclipse-ee4j/batch-tck

CC @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @scottmarlow

---

I think we are very close to adding a path covering the standalone platform tests in each of an EJB + Servlet context, as agreed on.   We're not quite there so I left this in 'draft' state.. but this is just an FYI that we're close.

### CODE REVIEW

**Comment by gurunrao** [comment]
@scottmarlow - If we go ahead with PR, we will not be able to run batch tests with Platform TCK EJB vehicle - https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/install/jakartaee/other/vehicle.properties#L264

**Comment by scottkurz** [comment]
> @scottmarlow - If we go ahead with PR, we will not be able to run batch tests with Platform TCK EJB vehicle - https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/install/jakartaee/other/vehicle.properties#L264

We expect to add an EJB path (as a variation to the existing Servlet path) within the platform portion of the standalone Batch TCK, implemented via Arquillian.    Tracked via: https://github.com/eclipse-ee4j/batch-tck/issues/22

**Comment by scottkurz** [comment]
@scottmarlow  - I think we should also add some small amount of documentation that says:  "To run the Platform, go run the platform portion of the Jakarta Batch  standalone TCK".   

Is there another example of where this is already done?     If not, I can take a shot at it?  Or would it be doc'd outside of this repo and more at some other, process level somewhere?

**Comment by scottmarlow** [comment]
> > @scottmarlow - If we go ahead with PR, we will not be able to run batch tests with Platform TCK EJB vehicle - https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/install/jakartaee/other/vehicle.properties#L264
> 
> We expect to add an EJB path (as a variation to the existing Servlet path) within the platform portion of the standalone Batch TCK, implemented via Arquillian. Tracked via: [eclipse-ee4j/batch-tck#22](https://github.com/eclipse-ee4j/batch-tck/issues/22)

:+1: 

IMO, we shouldn't remove the Batch tests from the Platform TCK until the work is complete to include an EJB path.  Thank you Scott for creating this pr for dealing the test removal, very much appreciated!

**Comment by scottmarlow** [comment]
> @scottmarlow - I think we should also add some small amount of documentation that says: "To run the Platform, go run the platform portion of the Jakarta Batch standalone TCK".

:+1: that sounds perfect to me.  We can always add more details about the Batch standalone TCK (e.g. link to the Batch TCK or perhaps to https://jakarta.ee/specifications/batch/2.1/ page which will later have the TCK link).

> 
> Is there another example of where this is already done? If not, I can take a shot at it? Or would it be doc'd outside of this repo and more at some other, process level somewhere?

Yes, https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/user_guides/jakartaee/src/main/jbake/content/intro.adoc contains a few relevant sections.

About what Platform TCK tests:
```
Technologies Tested with Jakarta EE 9.1 Platform TCK

The Jakarta EE 9.1 Platform TCK test suite includes compatibility tests for the
following required and optional Jakarta EE 9.1 technologies:

* Jakarta Enterprise Beans 4.0 +
* Jakarta Servlet 5.0
* Jakarta Server Pages 3.0
* Jakarta Expression Language 4.0
* Jakarta Messaging 3.0
* Jakarta Transactions 2.0
* Jakarta Mail 2.0
* Jakarta Connectors 2.0
* Jakarta Enterprise Web Services 2.0 (optional)
* Jakarta RESTful Web Services 3.0
* Jakarta WebSocket 2.0
* Jakarta JSON Processing 2.0
* Jakarta JSON Binding 2.0
* Jakarta Concurrency 2.0
* Jakarta Batch 2.0
* Jakarta Authorization 2.0
* Jakarta Authentication 2.0
* Jakarta Standard Tag Library 2.0
* Jakarta Faces 3.0
* Jakarta Security 2.0
* Jakarta Annotations 2.0
* Jakarta Persistence 3.0
* Jakarta Bean Validation 3.0
* Jakarta Managed Beans 2.0
* Jakarta Interceptors 2.0
* Jakarta Contexts and Dependency Injection 3.0
* Jakarta Dependency Injection 2.0
* Jakarta Debugging Support for Other Languages 2.0
* Jakarta Enterprise Beans 3.2 and earlier entity beans and associated Jakarta Enterprise Beans QL (optional)
* Jakarta Enterprise Beans 2.x API group (optional)
* Jakarta Enterprise Web Services 2.0 (optional)
* Jakarta SOAP with Attachments 2.0 (optional)
* Jakarta Web Services Metadata 3.0 (optional)
* Jakarta XML Web Services 3.0 (optional)
```

Additional Jakarta EE 9.1 Platform TCK Requirements:
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In addition to the instructions and requirements described in this
document, all Jakarta EE 9.1 and Jakarta EE 9.1 Web Profile implementations must
also pass the standalone TCKs for the following technologies:

* Jakarta Contexts and Dependency Injection 3.0
* Jakarta Dependency Injection 2.0
* Jakarta Bean Validation 3.0
* Jakarta XML Binding 3.0

For more information about the Jakarta Contexts and Dependency Injection technology, 
see the specification at `https://jakarta.ee/specifications/cdi/3.0/`

For more information about the Jakarta Dependency Injection,
see the specification at `https://jakarta.ee/specifications/dependency-injection/2.0/` 

For more information about the Jakarta Bean Validation technology, see the
specification at `https://jakarta.ee/specifications/bean-validation/3.0/`

For more information about the Jakarta XML Binding technology, see the
specification at `https://jakarta.ee/specifications/xml-binding/3.0/`

[[GBFQW]][[getting-started-with-the-jakarta-ee-9-platform-tck-test-suite]]
```

**Comment by scottmarlow** [comment]
> @scottmarlow marking for review, though I'd imagine I didn't get everything.
> 
> This is complicated by the fact that "batch" as in generic grouping is used in a number of places that are NOT referring to Jakarta Batch, preventing a simple find/replace.

Agreed.

> # Batch Standalone TCK ready
> 
> First, with the merge of: [eclipse-ee4j/batch-tck#40](https://github.com/eclipse-ee4j/batch-tck/pull/40) and the pending related doc PR: [eclipse-ee4j/batch-tck#41](https://github.com/eclipse-ee4j/batch-tck/pull/41) we do now have functional equivalence in that we are testing from both a servlet and Enterprise Bean context
> # Removed from platform TCK
> 
> As you can see I removed
> 
>     1. the Batch TCK tests themselves
> 
>     2. the Batch signature files
> 
>     3. references (that I saw) to batch/jbatch in the automation
> 
>     4. references (that I saw) to batch/jbatch in the doc
> 
> 

:+1: 

> ## Comment
> 
> I updated the entry in: user_guides/jakartaee/src/main/jbake/content/intro.adoc as mentioned but I noticed it's still at the EE 9.1 level of spec versions. There isn't some other branch I should be working off is there?
> # Question
> 
> One thing I was unsure about is whether to remove signature test related entries. We do have a full set of signature tests in the Batch Standalone TCK, so maybe we don't need them here. OTOH, I see CDI-related entries and CDI has its own sig tests, IIUC, so maybe they should be left in?

We will try to avoid duplication of tests that users run so I agree with removing the Jakarta Batch TCK signature test from the Platform TCK (as long as the Standalone Batch TCK signature map file is equivalent to the Jakarta Platform TCK Batch TCK signature map file).  

If for example you notice that the Batch TCK signature map file is missing some classes for a certain Jakarta EE SPEC API, that likely can be resolved by changing how you generate the signature map file to include the class package name for the incomplete SPEC API (for example of that type of issue being addressed, see https://github.com/eclipse-ee4j/cdi-tck/pull/348.)  

Note that the https://github.com/eclipse-ee4j/jakartaee-tck/tree/master/user_guides/jakartaee/src/main/jbake/content/intro.adoc is in need of an update to section 1.5 (`Additional Jakarta EE 9.1 Platform TCK Requirements`) but that doesn't need to change with this pull request (a lot of versions and text is pending to be updated in the Platform TCK user guide).  

@scottkurz could you please send email to the Platform, Platform TCK, Batch mailing lists asking for a review of this pull request for removal of duplicate Jakarta Batch tests?  Or ping me back if you would prefer that I send that, either way is fine.  

https://github.com/eclipse-ee4j/jakartaee-platform/wiki/Guidelines_For_Standalone_TCKs mentions that we want the different teams to have a chance to review changes that remove Platform TCK tests due to duplication between Standalone TCK + Platform TCK.

Thanks!
Scott

**Comment by scottkurz** [comment]
OK, updated PR removing the signature tests.  This is to the best of my ability as I can determine just by eyeballing the source;  I'm not verifying this by actually running anything. (I haven't done this previously or I would have tried.)   

I think I deleted all references to Jakarta Batch  and left all references to "batch testing" and "batch" APIs like JDBC "batch update".   But please check.

## QUESTION?
One place in particular I wasn't sure about was:  **user_guides/jakartaee/src/main/jbake/content/rules.adoc**  
I left it in since it seemed to be saying Batch was part of the Platform in a more abstract sense.  But maybe I'm wrong and this refers to a more specific test construct and so Jakarta Batch  should also be removed from this list.

**Comment by scottkurz** [comment]
@scottmarlow just noting we made it to the two week mark.  I don't see that there have been any comments. 

I have been rebasing/fixing the merge conflicts.   As I mentioned I'm not familiar enough to go ahead and try to test these changes (i.e. run the TCK w/o Batch).

File: docker/build_signatures.sh
Comment (scottmarlow) [inline docker/build_signatures.sh:61]: We should not update docker/build_signatures.sh as per my general comment about having Batch SPEC API signature checking in both Standalone + Platform TCK.

File: docker/pom.xml
Comment (scottmarlow) [inline docker/pom.xml:55]: We should leave docker/pom.xml as is also as that is used for generating signature map files.

File: src/com/sun/ts/tests/signaturetest/javaee/JavaEESigTest.java
Comment (scottmarlow) [inline src/com/sun/ts/tests/signaturetest/javaee/JavaEESigTest.java:88]: We should keep current src/com/sun/ts/tests/signaturetest/javaee/JavaEESigTest.java settings also for signature testing.

File: src/com/sun/ts/tests/signaturetest/recordSigs.sh.txt
Comment (scottmarlow) [inline src/com/sun/ts/tests/signaturetest/recordSigs.sh.txt:65]: I think this file is ignored, so either keep or revert the change to this file.

File: install/jakartaee/bin/sig-test-pkg-list.txt
Comment (scottmarlow) [inline install/jakartaee/bin/sig-test-pkg-list.txt:41]: It shouldn't matter if you keep or remove this change for signature testing as we are using docker/build_signatures.sh

File: install/jakartaee/bin/sig-test.map
Comment (scottmarlow) [inline install/jakartaee/bin/sig-test.map:160]: We should keep this change for running the signature testing with Platform TCK.

**Review by scottmarlow — CHANGES_REQUESTED** [review]
Looks good to me other than the requested changes requested above.

**Comment by scottkurz** [comment]
@scottmarlow I updated this PR adding back the signature test removals.  Thanks for the detailed comments.  I think I captured them correctly.   

In case you're trying to re-review, let me note that I left the first 3 commits in the PR and replaced only the 4th commit.

If you want to simply diff the new, updated PR (with the new 4th commit) with what I had before (but force pushed over) you can see the earlier copy at: 
https://github.com/scottkurz/jakartaee-tck/tree/remove-batch-tck-prev

I didn't simply revert the 4th commit because I had a couple things besides sigtest stuff in there.

Thanks.

**Comment by scottmarlow** [comment]
I'll do some testing on Jenkins of this change this week.

**Review by scottmarlow — APPROVED** [review]

**Comment by scottkurz** [comment]
Thx @scottmarlow  - just reminding you I'm not a committer able to merge the PR myself.

**Comment by scottmarlow** [comment]
> Thx @scottmarlow - just reminding you I'm not a committer able to merge the PR myself.

Thanks, just giving others a change to review also before merging.

**Comment by gurunrao** [comment]
@scottkurz @scottmarlow  - trying to get test run of platform TCK with the PR changes, blocked by Platform TCK build issues.

**Comment by scottmarlow** [comment]
Deferring the merge until after we fix Platform TCK build

**Comment by scottmarlow** [comment]
Started https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck-scottmarlow/job/remove-batch-moved-to-standalone-tck/2/ to verify that we can still build the Platform TCK after merging this change.

**Pull request merged**
